src/librustc/ty/print/pretty.rs

```diff
@@ -851,50 +864,67 @@ pub trait PrettyPrinter<'tcx>:
                         if let Ok(snip) = self.tcx().sess.source_map().span_to_snippet(span) {
                             p!(write("{}", snip))
                         } else {
-                            p!(write("_: "), print(ct.ty))
+                            p!(write("_: {}", ct.ty))
                         }
                     } else {
-                        p!(write("_: "), print(ct.ty))
+                        p!(write("_: {}", ct.ty))
```

Comment:
I don't get these changes - this is *never* correct, because it takes away control from the current pretty printer, and moves it to the default `fmt::Display` one.

Maybe I didn't mean to suggest `type_ascribed`(`_value`) to have braces? I really can't remember though. If you move the braces to its users you can use it here too.

Reply:
I reverted all of these